src/shared/__tests__/extract-error-string.test.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import {extractErrorString} from "../extract-error-string.js";
+
+describe("#extractErrorString", () => {
+    it("should return the error if it is a string", () => {
+        // Arrange
+        const error = "THIS IS AN ERROR!";
+
+        // Act
+        const result = extractErrorString(error);
+
+        // Assert
+        expect(result).toBe(error);
+    });
+
+    it.each([
+        ["THIS IS A STACK!", "THIS IS THE ERROR!: THIS IS A STACK!"],
+        [undefined, "THIS IS THE ERROR!: "],
+    ])(
+        "should return string if error has response property with error property",
+        (stack, expectation) => {
+            // Arrange
+            const error = {
+                response: {
+                    error: "THIS IS THE ERROR!",
+                },
+                stack,
+            };
+
+            // Act
+            const result = extractErrorString(error);
+
+            // Assert
+            expect(result).toBe(expectation);
+        },
+    );
+
+    it("should recurse if error has error child", () => {
+        // Arrange
+        const error = {
+            error: {
+                error: {
+                    error: "THIS IS THE ERROR!",
+                },
+            },
+        };
+
+        // Act
+        const result = extractErrorString(error);
+
+        // Assert
+        expect(result).toBe("THIS IS THE ERROR!");
+    });
+
+    it("should return the stack if it exists", () => {
+        // Arrange
+        const error = {
+            stack: "THIS IS THE STACK!",
+        };
+
+        // Act
+        const result = extractErrorString(error);
+
+        // Assert
+        expect(result).toBe("THIS IS THE STACK!");
+    });
+
+    it("should return the result of toString if none of the other cases match", () => {
+        // Arrange
+        const error = {
+            this: "IS NOT LIKE THE OTHERS",
+            toString: () => "AND SO THIS IS THE ERROR!",
```

Comment:
Using `this` as a key here isn't necessary is it?  It's a joke right?

Reply:
Yeah, it's just a joke :grin:

Reply:
It was funny, but seeing `this` as a key got my spidey sense tingling.  Please don't take the fact that I asked if it was a joke as it being unfunny. 😅

Reply:
Can you imagine typing `this.this`? Imagine reviewing that code. lol